### PR TITLE
dotCMS/core#19701 Add the "don't show again" checkbox

### PR DESCRIPTION
**Pull request by hmoreras**
- Add Checkbox

### CODE REVIEW

File: src/app/api/services/dot-starter/dot-starter.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { CoreWebService } from 'dotcms-js';
+
+@Injectable()
+export class DotStarterService {
```
Comment (fmontes) [inline src/app/api/services/dot-starter/dot-starter.service.ts:5]:
```suggestion
export class DotToolGroupStarterService {
```

Right now it looks like a service to handle "starters" and is not that.

Don't forget file names.

File: src/app/api/services/dot-starter/dot-starter.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    constructor(private coreWebService: CoreWebService) {}
+
+    hide(): any {
```
Comment (fmontes) [inline src/app/api/services/dot-starter/dot-starter.service.ts:8]: doc and type

File: src/app/api/services/dot-starter/dot-starter.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        return this.coreWebService.requestView({
+            url: 'api/v1/toolgroups/gettingstarted/_removefromcurrentuser',
+            method: 'PUT'
+        });
+    }
+
+    show(): any {
```
Comment (fmontes) [inline src/app/api/services/dot-starter/dot-starter.service.ts:15]: doc and type

File: src/app/portlets/dot-starter/dot-starter.component.ts [hunk 4]
```diff
@@ -10,11 +11,18 @@ import { pluck, take } from 'rxjs/operators';
...
+    constructor(private route: ActivatedRoute, private dotStarterService: DotStarterService) {}
 
     ngOnInit() {
         this.route.data.pipe(pluck('username'), take(1)).subscribe((username: string) => {
             this.username = username;
         });
     }
+
+    handleVisibility(hide: boolean): void {
+        console.log(hide);
```
Comment (fmontes) [inline src/app/portlets/dot-starter/dot-starter.component.ts:23]: remove

File: src/app/portlets/dot-starter/dot-starter.component.ts [hunk 5]
```diff
@@ -10,11 +11,18 @@ import { pluck, take } from 'rxjs/operators';
...
-    constructor(private route: ActivatedRoute) {}
+    constructor(private route: ActivatedRoute, private dotStarterService: DotStarterService) {}
 
     ngOnInit() {
         this.route.data.pipe(pluck('username'), take(1)).subscribe((username: string) => {
             this.username = username;
         });
     }
+
+    handleVisibility(hide: boolean): void {
```
Comment (fmontes) [inline src/app/portlets/dot-starter/dot-starter.component.ts:22]: doc.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: src/app/api/services/dot-tool-group/dot-tool-group.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        expect(req.request.method).toBe('PUT');
+    });
+
+    it('should call the API to show the portlet', () => {
+        const url = 'api/v1/toolgroups/gettingstarted/_addtocurrentuser';
+        service.show('gettingstarted').subscribe();
+
+        const req = httpMock.expectOne(url);
+        expect(req.request.method).toBe('PUT');
+    });
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-tool-group/dot-tool-group.service.spec.ts:38]: you are missing
``` afterEach(() => {
        httpMock.verify();
    });```

**Review by fmontes — APPROVED** [review]

**Pull request merged**